### PR TITLE
pcrphase: add two additional phases

### PR DESCRIPTION
**Pull request by poettering**
This adds two more phases to the PCR boot phase logic: "sysinit" + "final".

The "sysinit" one is placed between sysinit.target and basic.target. It's good to have a milestone in this place, since this is after all file systems/LUKS volumes are in place (which sooner or later should result in measurements of their own) and before services are started (where we should be able to rely on them to be complete).

This is particularly useful to make certain secrets available for mounting secondary file systems, but making them unavailable later.

This breaks API in a way (as measurements during runtime will change), but given that the pcrphase stuff wasn't realeased yet should be OK.

### CODE REVIEW

**Comment by poettering** [comment]
This is a slight extension of the earlier "pcrphase" commit, but makes the phases a bit more finegrained. Given this is somewhat API relevant, let's add this before the 252 release.

This adds no new code, just a new unit for calling systemd-pcrphase.

**Comment by poettering** [comment]
Follow-up for 708d7524790c962a5b49bcf8bc5556f4acda3d08 btw

File: src/boot/measure.c [hunk 1]
```diff
@@ -281,7 +281,8 @@ static int parse_argv(int argc, char *argv[]) {
                 if (strv_extend_strv(&arg_phase,
                                      STRV_MAKE("enter-initrd",
                                                "enter-initrd:leave-initrd",
-                                               "enter-initrd:leave-initrd:ready"),
+                                               "enter-initrd:leave-initrd:sysinit",
+                                               "enter-initrd:leave-initrd:sysinit:ready"),
```
Comment (daandemeyer) [inline src/boot/measure.c:285]: Isn't this missing `final`?

Comment (poettering) [reply src/boot/measure.c:285]: no. By default we'll generate encrypted secrets you can unlock from initrd on all the way to the beginning of shutdown. but not before the initrd and not after shutdown began.

With other words the boot path `:`, `enter-initrd:leave-initrd:sysinit:ready:shutdown`,  `enter-initrd:leave-initrd:sysinit:ready:shutdown:final` are not covered, on purpose.

The idea is that during bootup and runtime people expect to be able to unlock secrets. But during shutdown? Probably not (or if they do, they should explicitly specify that). It's generally a good thing that secrets cannot be unlocked anymore during shutdown, for example if for some reason shutdown hangs or so.

**Review by bluca — APPROVED** [review]

**Pull request merged**